src/core/blocks/base_block.c

```diff
@@ -0,0 +1,59 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "base_block.h"
+
+char * ser_BlockHeader(BlockHeader *block_header, char* dest){
```

Comment:
Could we pick a function naming convention to use? I'd prefer using only camel case or snake case (SerBlockHeader or ser_block_header?) instead of mixing.